src/components/ComponentSettingsModal.jsx

```diff
@@ -0,0 +1,467 @@
+                        onFieldChange={this.changeHandler}
+                      />
+                    </div>
+                    
+                    <div className="col-8">
+                      <p className={elementStyles.formLabel}>Resource Type</p>
+                      {/* Permalink or File URL */}
+                      <div className="d-flex">
+                        <label htmlFor="radio-post" className="flex-fill">
+                          <input
+                            type="radio"
+                            id="radio-post"
+                            name="resource-type"
+                            value="post"
+                            onClick={this.handlePermalinkFileUrlToggle}
+                            checked={isPost}
+                          />
+                          Post Content
+                        </label>
+                        <label htmlFor="radio-post" className="flex-fill">
+                          <input
+                            type="radio"
+                            id="radio-file"
+                            name="resource-type"
+                            value="file"
+                            onClick={this.handlePermalinkFileUrlToggle}
+                            checked={!isPost}
+                          />
+                          Downloadable File
+                        </label>
+                      </div>
+                    </div>
+                  </div>
+                  {/* File URL */}
+                  <FormFieldFile
+                    title="Select File"
+                    id="fileUrl"
+                    value={fileUrl?.split('/').pop()}
+                    errorMessage={errors.fileUrl}
+                    onFieldChange={this.changeHandler}
+                    inlineButtonText="Select File"
+                    disabled={isPost}
+                    onInlineButtonClick={() => this.setState({ canShowFilesModal: true })}
+                  />
+                </>}
+              </div>
+              {/* Save or Delete buttons */}
+              <div className={elementStyles.modalButtons}>
+                <LoadingButton
+                  label="Save"
+                  disabled={hasErrors}
+                  disabledStyle={elementStyles.disabled}
+                  className={`ml-auto ${(isNewFile ? hasErrors : (hasErrors || !sha)) ? elementStyles.disabled : elementStyles.blue}`}
+                  callback={this.saveHandler}
+                />
+                { !isNewFile
+                  ? (
+                    <LoadingButton
+                      label="Delete"
+                      disabled={(hasErrors)}
+                      disabledStyle={elementStyles.disabled}
+                      className={`${hasErrors ? elementStyles.disabled : elementStyles.warning}`}
+                      callback={() => this.setState({ canShowDeleteWarningModal: true })}
+                    />
+                  ) : null}
```

Comment:
I think the buttons can be their own component as well since they're used in quite a few places - what do you think?

Reply:
Resolved in https://github.com/isomerpages/isomercms-frontend/pull/184/commits/daa5e79a1ccb17d83c929cdf856b387a739a2ff3